### DISCUSSION TITLE
FUNCTION_INVOCATION_TIMEOUT with Firebase uploadBytes() on Vercel

### DISCUSSION THREAD

**Discussion by XiiinSheng** [vercel/community#1292] [Help] [answered]
### Summary

I'm building an Express server that receives and uploads small mp3 files to Firebase. 

The functionalities were working fine on Vercel until 2 months ago. Since that time all the `uploadBytes()` functions timeout after stucking at 0% upload for 10.01s. The function works well on localhost, but not on Vercel. 

### Additional information

_**Error Output**_
Function Logs on Vercel (with some console.log outputs):
![image](https://user-images.githubusercontent.com/57552802/213276400-c7ff51bc-5638-4db0-9222-4136dcec5f5b.png)
Error Output in Postman:
![image](https://user-images.githubusercontent.com/57552802/213277428-a1eeaa5b-3726-4ef4-88bb-ff1b1397e662.png)

_**Code Debug**_
Using `uploadBytesResumable()` I saw that the upload process is stuck at 0% with the output like this: 
![image](https://user-images.githubusercontent.com/57552802/213278118-f1d48dc7-baa1-47ae-b57a-290223f08cf3.png)

**_Vercel Reverting Attempt_**
I have a working deployment, so I tried to deploy the exact same code again, but the problem exists.  
I also tried to promote this deployment back to production, but the problem still exists.  
The only working option is to "instant rollback" to that deployment, and the code works fine. 
Any new deployment produces the problem.

**Comment by amyegan** [comment]
Can you share links to the working and failing deployments or a [minimal reproduction](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)? That will help us get a better look at what is going wrong with the upload.

**Comment by XiiinSheng** [comment, marked as answer]
Problem is fixed by updating Firebase module from ^9.9.2 to 9.13.0. Possibly a compatibility issue between Vercel and Firebase. 
Thank you for helping!

**Reply by NicolaGenesinJetty** [reply]
This fixed it for me as well. Thank you!
